docs/developer-guide/attributes.md

```diff
@@ -85,26 +133,39 @@ In addition, normalization, maps values stored in an integer format to a normali
 * `[-1,1]` (SNORM, for signed integers)
 * `[0,1]` (UNORM, for unsigned integers)
 
-In WebGL2, it is possible to disable automatic conversion of integers to integers, enabling shaders to work directly with integer values. This works with all the integer types: `gl.BYTE`, `gl.UNSIGNED_BYTE`,
-`gl.SHORT`, `gl.UNSIGNED_SHORT`, `gl.INT` and `gl.UNSIGNED_INT`.
+In WebGL2, it is possible to disable automatic conversion of integers to integers, enabling shaders to work directly with integer values. This works with all the integer types: `gl.BYTE`, `gl.UNSIGNED_BYTE`, `gl.SHORT`, `gl.UNSIGNED_SHORT`, `gl.INT` and `gl.UNSIGNED_INT`.
 
 
-### WebGL2 Changes
+### Transform Feedback
 
-> The differences described here are hidden by the luma.gl `VertexArray` API.
+ For transform feedback operations, the `TransformFeedback` class holds the output buffers that will receive data from vertex shader varyings, complementing the `VertexArray` class which holds the input values. It is worth noting that the `TransformFeedback` class is quite similar to the `VertexArray` class in concept and API, they both read metadata from `Program` instances, etc.
 
-The raw WebGL APIs for `WebGLVertexArray`s are exposed differently in the WebGL1 extension and WebGL2. As always, the luma.gl `VertexArray` class transparently handles the necessary API detection and selection.
 
-**`ANGLE_instanced_arrays` Extension** Allows instance divisors to be set, enabling instanced rendering.
-* **`OES_VertexArray` Extension** Enables the application to create and "VertexArray"s to save and restore the entire global vertex attribute array with a single operation. luma.gl provides a class wrapper for `VertexArrays`.
+## Background: How the GPU Accesses Attributes
 
-* Setting instance divisors no longer requires a WebGL extension.
-* `VertexArrays` no longer require using a WebGL extension.
-* Adds support for exposing integer attribute values directly to shaders (without those values first being auto-converted to floats) The improvements cover both constant and buffer-valued attributes.
+GPUs shaders perform parallel processing on "vertices", and one "value" in each attribute is made available to the particular shader invocation that processes the corresponding "vertex".
+
+At the start of shader execution, these indices (or 'locations') are matched with small integer indices assigned to shader attributes during shader compilation and program linking. This makes the data the application has set up in the vertex attributes available during shader execution. Vertex attributes thus represent one of the primary mechanisms for communication between JavaScript code and GPU code (GLSL shaders).
```

Comment:
May be rewrite this in following way or adjust this a bit?
`During the shader compilation and linking, GPU's predefined range of indices (or `locations`), typically 0 to 16,  are mapped to the inputs (attributes) of the vertex shader. Application can then bind `Attributes` to these mapped indices, hence `Attributes` represent one of the ...`